src/build/prepareElectronApp.ts

```diff
@@ -126,21 +127,29 @@ async function maybeCopyScripts(srcs: string[], dest: string): Promise<void> {
     await copyFileOrDir(src, destPath);
   }
 }
-
-function normalizeAppName(appName: string): string {
-  // use a simple random string to prevent collisions
-  const postFixHash = generateRandomSuffix();
+/**
+ * Use a simple 3-character hash to prevent collisions. The hash is deterministic-by-url,
+ * so that an upgrade (same URL) of an existing app keeps using the same appData folder.
+ */
+function normalizeAppName(appName: string, url: string): string {
+  const hash = crypto.createHash('md5');
```

Comment:
I think I have a solution for this that is best of both worlds if you're interested in me applying it here.

Reply:
@TheCleric what's the solution?

Reply:
Change the `generateRandomSuffix` function to accept an optional seed. So here we could pass the url and get the same behavior, but for the JS code, just let it use a random one if it wasn't provided (as it is now).

Reply:
> Change the `generateRandomSuffix` function to accept an optional seed. So here we could pass the url and get the same behavior, but for the JS code, just let it use a random one if it wasn't provided (as it is now).

@TheCleric hmmmm we could, but I'd rather let `normalizeAppName` and `generateRandomSuffix` be different functions.

They both do hashing, but have different needs, and there's a kind of backwards compatibility we help preserve by inlining `normalizeAppName`'s hashing. By making `normalizeAppName` use `generateRandomSuffix`, we're suggesting it can use a generic/reusable thing, which is not true: for backwards compat of appData folder we do want `normalizeAppName` to remain stable. So we don't want to extract this specific hashing further away in a function, where it'll be at the mercy of a future refactor that will fall into the same gotcha, we want to keep it close/inlined.

Reply:
@TheCleric actually, adding a test checking `normalizeAppName` stability